blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java

```diff
@@ -260,13 +260,17 @@ protected void handleChunkDone(@Nonnull MemoryFlowChunk chunk) {
             }
         }else{
             if(parallelNestedStages) {
-                String endId = parallelBranchEndNodes.peek().getId();
-                Stack<FlowNodeWrapper> stack = stackPerEnd.get(endId);
-                if(stack==null){
-                    stack=new Stack<>();
-                    stackPerEnd.put(endId, stack);
+                if(parallelBranchEndNodes.isEmpty()){
+                    logger.warn("skip parsing stage {} but parallelBranchEndNodes is empty", stage);
```

Comment:
Do we want this to be warning or debug level? In which condition we will hit this case? Making sure what this warning level means in terms of identifying underlying issue.

Reply:
oops. I added as warning level to ease debug with @NicuPascu . Sure debug might be better.
Keep as it and I will change that once everything is ok for @NicuPascu.
So we can wait before merging this pr.